### ISSUE TITLE
Refunded order gets wrong state.

### ISSUE THREAD

**Issue by jacob-v-dam** [mollie/PrestaShop#249]
### Expected behavior
When an order is refunded the order gets the state "paid" I checked the configuration but I would expect that it should go to another option like this:
![image](https://user-images.githubusercontent.com/310526/100381601-99f1ac00-3019-11eb-94b9-034a66420e67.png)

### Actual behavior
Order gets state paid:
![image](https://user-images.githubusercontent.com/310526/100381627-a8d85e80-3019-11eb-9907-1c5ca1382a9b.png)

This happend to iDeal orders, don't know if this happens with order payment methods.

### Steps to reproduce the behavior
Create an iDeal order. Pay this order and refund it.

### Environment
* PrestaShop version: 1.7.6.9
* Mollie module version: 4.1.1
* PHP version: 7.2.34

**Comment by Maz-Invertus** [comment]
Hi @jacob-v-dam !

Thanks a lot for your input to the Mollie community. 
Just wanted to double check have you noticed this behavior only on 4.1.1 version? or previous versions too?

Looking forward to hearing from you

**Comment by jacob-v-dam** [comment]
This also happend in version 4.1.0 therefor i updated to version 4.1.1 to see if the problem still existed, unfortunately it did.

**Comment by SimonasB88** [comment]
Hello @jacob-v-dam thank you for your insights,

We have tested on our testing environments, we couldn't find any code problems, seems that Mollie has a little bit changed the Refund functionality in the Mollie.com dashboard.

If you could check your Mollie account, you could see that now you can cancel the Refund action, and the Refund status itself is pending. The module is mirroring the same statuses at this functionality as the Mollie dashboard does.

![image](https://user-images.githubusercontent.com/29095235/100608183-b725c380-3314-11eb-8c25-c7658e4bc50e.png)

However, if you still experiencing the problem, that could be not related to this Mollie update, could check and write us here?
Or could we check your webshop by logging in and debugging there? 

Best Regards.
support@invertus.eu

**Comment by thezimba** [comment]
We have the same problem, it seems that the Mollie backend triggers a statusupdate to the status "Paid".

Can you please look in to this?

**Comment by Saimis777** [comment]
@thezimba Hey, are you still encountering the issue? If so, would you be able to create a separate issue for it? Or write us at - support@invertus.eu for further investigation!

--
Best regards,
Invertus Support team

**Comment by jacob-v-dam** [comment]
I'm still encountering this issue. As soon as the refund is requested the state goes back to 'Betaling aanvaard', which is paid.

If you change your order to a different state and request a refund the error occurs, it goes back to 'Betaling aanvaard'.

**Comment by Saimis777** [comment]
Hello @jacob-v-dam, thank you for your message!

We've tested this on an https secure connection, and it seems that the refund is done appropriately (adding screenshots)
![image](https://user-images.githubusercontent.com/75423142/102515063-f171bd80-4095-11eb-9250-deef78ea793d.png)
![image](https://user-images.githubusercontent.com/75423142/102515087-f9c9f880-4095-11eb-91a6-0ddfc81384e2.png)

Maybe you could screenshot the webhook transactions on the right side of the dashboard? The errors can be seen here:
![image](https://user-images.githubusercontent.com/75423142/102515353-5200fa80-4096-11eb-88c5-10cf020b8a0f.png)

If we could see the errors, it would be easier to identify the issue!

Looking forward to your reply!

Best regards,
Invertus Support team

**Comment by jacob-v-dam** [comment]
There are no errors in the refunded order. I create the refund from the mollie table, could this be the problem? And most of the time it as an iDeal payment.

**Comment by Saimis777** [comment]
@jacob-v-dam Could you maybe clarify if you're testing this on a localhost? And after doing the refund, does the status not change correctly even after 10-15 minutes after making the refund?

Looking forward to your reply!

Best regards,
Invertus Support team

**Comment by Saimis777** [comment]
Hello @jacob-v-dam, how are you?

Have you managed to sort out the issue? Or does it still persists? If yes, would you be able to clarify the needed information from my last message for further investigation?

Kindly looking forward to your reply!

Best regards,
Invertus Support team

**Comment by jacob-v-dam** [comment]
I made a refund and happend again. Again it was a iDeal payment.

The refund was made possible with the table on the order detail page.

**Comment by Saimis777** [comment]
@jacob-v-dam And you're testing this on a local host? Or the page is live on a hosting server?

Looking forward to your reply!

Best regards,
Invertus Support team

**Comment by jacob-v-dam** [comment]
This is a live server. Probably you could fix the log reporting and then it would show up in the presta log

**Comment by Saimis777** [comment]
@jacob-v-dam Could you maybe send us your Front office/ Back office + FTP + clear steps how to replicate the issue to - support@invertus.eu for further investigation? As at the moment it's difficult to determine what exactly is wrong, since we couldn't replicate it from our end.

Best regards,
Invertus Support team

**Comment by Saimis777** [comment]
Hey @jacob-v-dam I've double checked this issue further, and came to the conclusion, that this is not really an issue, as I've got a full clarification, that the official time that the status should change is within 2 hours, so please let me know if the status doesn't change even after 2 hours, if it does, then it's normal, if it doesn't change even after 2 hours mark, then it's an issue.

Looking forward to your reply!

Best regards,
Invertus Support team

**Comment by jacob-v-dam** [comment]
I'll try to provide FTP credentials this week.

But I think I have found something else:

**Order complete**
When someone returns an order and it gets refunded everything goes well. The order is marked as shipped in Mollie and it doesn't get the paid state.

**Order cancelled**
When an order is cancelled but it isn't delivered it gets the paid state when it is refunded. In my case an order gets a different state for order handling. So the paid state is not set anymore this allows the module to set the paid state again and send an e-mail.

**Comment by Saimis777** [comment]
Thank you for that, would be best to provide back office/ front office login details as well, as we couldn't reproduce the issue from our end + clear steps how to encounter it, so we could rapidly check how to solve it, if the issue can actually be reproduced.

Be sure to send the login details at - support@invertus.eu

Best regards,
Invertus Support team

**Comment by basemedia** [comment]
Is it possible to remove the 2 hour check?

**Comment by Saimis777** [comment]
Hello @basemedia, thank you for your message! But unfortunately the 2 hour mark is default, and intended Mollie behaviour, therefore it can't be changed, sorry for the inconveniences.

Let me know if you have any further questions!

Best regards,
Invertus Support team

**Comment by basemedia** [comment]
That's no problem. I understand. Is it possible to disable the Webhook call? I am using module 3.5.5.

Let me know.

**Comment by Saimis777** [comment]
@basemedia Yeah, it's possible but, the functionality of returning the statuses won't work, when the webhook call is disabled. I would highly suggest firstly doing this on a test environment, only then actually proceeding with disabling it, but most likely it could break the module.

Let me know if you have any other questions, we are here to help!

Best regards,
Invertus Support team

**Comment by basemedia** [comment]
Ok. Please inform how I can disable the call.

**Comment by Saimis777** [comment]
Hello,

To disable the webhook you can comment the whole file in order to disable it, the webhook can be found - controllers/front/webhook.php, or in the function "initContent" you can perform die('ok'); and that should disable it.

Best regards,
Invertus Support team

**Comment by Saimis777** [comment]
Hello @jacob-v-dam does your issue still persists? Would you be able to provide the credentials for further checking?

Also, 4.2.0 version was just released, you can download it here - https://github.com/mollie/PrestaShop/releases/tag/4.2.0

Any feedback regarding the new version is greatly appreciated! 

Looking forward to your reply!

Best regards,
Invertus Support team
